app/javascript/lib/imperial.js

```diff
@@ -32,16 +30,12 @@ import AvailableSlots from './UseCases/Rondels/SlotSelection/AvailableSlots';
 import SlotDistanceCosts from './UseCases/Rondels/SlotSelection/SlotDistanceCosts';
 
 export default class Imperial {
-  static fromLog(log, board) {
-    const game = new Imperial(board);
-    log.forEach((entry) => game.tick(entry));
-    return game;
-  }
+  #logger;
+  #game;
 
-  constructor(board) {
-    this.logtail = new Logtail('3bdHcA8P3mcww2ojgC5G8YiT');
+  constructor(board, logger) {
+    this.#logger = logger;
```

Comment:
What happens if we don't pass in a logger?

Reply:
How about if we change this to:

```javascript
this.#logger = logger || new Logger();
```

and this should fall through to the `default` case in the env check on `Logger.js`
```javascript
switch (this.#environment) {
  ...
  default:
    console.error(message, data);
    break;
}
```
Then we can remove this from the tests. Would that work?

Reply:
>do we want it to log in the tests tho, since it already logs on CI?

Not sure I understand. This would log locally in tests, which is probably a good thing, right? Like, if a test is passing but encounters an error anyway, it would be good to see that in the terminal output. Not sure what you mean by it already logs on CI.

>Or maybe the default case should be a silent fail, altho Im not as keen on silent failures

I feel the same way but I'm ok with it in this case if that seems better than `console.error`ing as above. Theoretically we shouldn't be seeing errors on tests that are passing. But you never know...!

Reply:
I see, I've never ran the tests locally, but yeah in that case it'd be more beneficial than detrimental, lets go with this plan than

Reply:
[76f61f0](https://github.com/Thrillberg/imperial/pull/729/commits/76f61f00f58e876e15be9e5e2d7e0ddfdbd8a432)